src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchive.Create.Unix.cs

```diff
@@ -10,7 +10,7 @@ public static partial class ZipFileExtensions
             Interop.Sys.FileStatus status;
             Interop.CheckIo(Interop.Sys.FStat(fs.SafeFileHandle, out status), fs.Name);
 
-            entry.ExternalAttributes |= status.Mode << 16;
+            entry.ExternalAttributes = status.Mode << 16;
```

Comment:
The ExternalAttributes store more things than just permissions, right? That's why we are bit-shifting the Mode 16.

So this is overwriting all the other bits already set in ExternalAttributes (if any).

Reply:
This seems correct, yes. My logic was this:

1) The default external attributes are `0` right now and I am going to change them to be `644` for files and `755` for folders (also `<< 16`) when running on a Unix system.
2) Since the default values are going to be set, we need to ensure that when somebody is using the API that gets the actual permissions from the files, gets the permissions that the files have.
3) This method, `SetExternalAttributes` is private (because it doesn't have a modifier on it) and is only used in `DoCreateEntryFromFile`. And it's called right after we create an entry.

So to me it seems like we are not overwriting anything there when we don't have to. Am I missing something?

Reply:
@eerhardt  I like the idea.
What would be your suggestion regarding making this const accessible in the `System.IO.Compression.ZipFile` assembly? Right now it's defined as internal and resides in `System.IO.Compression`.
Should we use `InternalsVisibleTo`? Or should we just make it public? Or any other options?

Reply:
I would make an internal shared source file that is compiled into both assemblies.

Reply:
@eerhardt Can you please check if this is how you wanted it?

Reply:
> so I don't find the assert necessary

My intention with the assert is to catch the case in the future where some new code gets added that is setting other pieces of the ExternalAttributes field. If there is ever a case where other pieces of ExternalAttributes is set, and then this method is called, we will be overwriting that information. The assert is there to alert that future developer that this code may need to change - for example, in that case it could just clear the permission bits first, and then `|= status.Mode << 16`

> the name SetExternalAttributes

Maybe a better name for this method would be `SetPermissions` or `SetFileMode`.

Reply:
Do you guys want me to change the name or should we keep it for now? Im okay with both.
Also, I think we might keep the assert: it doesn't really break anything. I like being extra careful.